research/conditional/utils/model_utils.py

```diff
@@ -245,6 +248,44 @@ def get_expert_choice_args(args):
     }
 
 
+def get_expert_choice_with_parallel_args(args):
```

Comment:
To reviewers: please verify carefully this function. This function is meant to split the compute between the expert layer and FF layer in a way that the total compute is always constant (independently of args.ff_parallel_compute_fraction).